internal/usage/aws/ec2.go

```diff
@@ -0,0 +1,37 @@
+package aws
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
+)
+
+func ec2NewClient(ctx context.Context, region string) (*ec2.Client, error) {
+	cfg, err := getConfig(ctx, region)
+	if err != nil {
+		return nil, err
+	}
+	return ec2.NewFromConfig(cfg), nil
+}
+
+func EC2DescribeOS(ctx context.Context, region string, ami string) (string, error) {
+	client, err := ec2NewClient(ctx, region)
+	if err != nil {
+		return "", err
+	}
+	result, err := client.DescribeImages(ctx, &ec2.DescribeImagesInput{
+		ImageIds: []string{ami},
+	})
+	if err != nil {
+		return "", err
+	} else if len(result.Images) == 0 {
+		return "", nil
+	}
+	switch result.Images[0].Platform {
```

Comment:
Nice start!  Just a heads up that I think we can get a better idea of the billing os by looking at the `UsageOperation` field instead of Platform.  There are some details sketched out in #778 and an example of the RunInstance mappings are https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/billing-info-fields.html

Reply:
Wow; that's deep. I'll skim the relevant API docs before I enshrine anything with tests.

Reply:
Good call; UsageOperation lets us infer RHEL and SUSE.

There are also "BYOL" variants of some usage classes, presumably cheaper since you pay the OS vendor directly vs. paying AWS. There is a `licenseModel` attribute query in CPAPI which I suspect is related, but I haven't delved into it at all yet. I'll leave it alone for now when building the cost component, but if you have any guidance LMK.